aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp

```diff
@@ -70,9 +70,10 @@ class QLinearDynamicInt8 final : public torch::OperatorKernel {
     auto q_params = quant_utils::ChooseQuantizationParams(
         /*min=*/x_min,
         /*max=*/x_max,
-        /*qmin=*/is_signed ? -(1 << (precision - 1)) : 0,
+        /*qmin=*/is_signed ? -(1 << (precision - 1 - 1)) : 0,
```

Comment:
We need to change the API for the op:
```python
      quantized::linear_dynamic(X, W_prepack)
```
to 
``` 
quantized::linear_dynamic(X,W_prepack, quant_min=-128, quant_max=127)
```
And in the dynamic/linear module:
Change forward from:
``` python
   def forward(self, x):
        # Note that we can handle self.bias == None case.
        if self._packed_params.dtype == torch.qint8:
            Y = torch.ops.quantized.linear_dynamic(
                x, self._packed_params._packed_params)
        elif self._packed_params.dtype == torch.float16:
            Y = torch.ops.quantized.linear_dynamic_fp16(
                x, self._packed_params._packed_params)
        else:
            raise RuntimeError('Unsupported dtype on dynamic quantized linear!')
        return Y.to(x.dtype)
```
To
```
   def forward(self, x):
        # Note that we can handle self.bias == None case.
        if self._packed_params.dtype == torch.qint8:
            Y = torch.ops.quantized.linear_dynamic(
                x, self._packed_params._packed_params, quant_min = self.quant_min, quant_max= self.quant_max)
        elif self._packed_params.dtype == torch.float16:
            Y = torch.ops.quantized.linear_dynamic_fp16(
                x, self._packed_params._packed_params)
        else:
            raise RuntimeError('Unsupported dtype on dynamic quantized linear!')
        return Y.to(x.dtype)
```
where we pass quant_min and quant_max as arguments to the init of the module, with default values of 0 and 255.

In
https://github.com/pytorch/pytorch/blob/812b1ad8695c7bf28a8caa2f001f0ec3c554fbb7/torch/nn/quantized/dynamic/modules/linear.py#L85, we do
``` python
        qlinear = Linear(mod.in_features, mod.out_features, dtype=dtype, quant_min=0, quant_max=127)
```
This way, existing models that are already scripted and deployed will not see a change in numerics and newer models can benefit from improved accuracy.

Reply:
Yep, mechanism like that is needed for BC. Please add a test for it too! We can either add explicit range or make it `bool reduce_range`. Explicit range is probably better

Reply:
Yes, there is a PR in the works to remove reduce range and make it explicit